pkg/resources/data.go

```diff
@@ -694,3 +695,97 @@ func (d *TemplateData) Seed() *kubermaticv1.Seed {
 func (d *TemplateData) KubermaticConfiguration() *kubermaticv1.KubermaticConfiguration {
 	return d.config
 }
+
+func GetEnvVars(data *TemplateData) ([]corev1.EnvVar, error) {
```

Comment:
```suggestion
func (d *TemplateData) GetEnvVars() ([]corev1.EnvVar, error) {
```

then you can:
- add `GetEnvVars() ([]corev1.EnvVar, error)` to the machinecontrolerdata and usercontrollerdata intefaces
- use `data.GetEnvVars()` instead of  `resources.GetEnvVars(data.(*resources.TemplateData))`